internal/admission/validator.go

```diff
@@ -630,3 +632,55 @@ func (m *managerClientConsumerGetter) ListAllConsumers(ctx context.Context) ([]k
 	}
 	return consumers.Items, nil
 }
+
+func (validator KongHTTPValidator) ValidateCustomEntity(ctx context.Context, entity kongv1alpha1.KongCustomEntity) (bool, string, error) {
+	// If the spec.contollerName does not match the ingress class name,
+	// ignore it as it is not controlled by the controller.
+	// REVIEW: should we also match the "ingress class" annotation here?
```

Comment:
Probably, unless it's particularly annoying to support both.

Most everything else we have honors the annotation, and consistency across the board is good UX. We're kinda split between Ingress and Gateway API conventions at present, but IMO GWAPI is still novel enough and not universally available that we wouldn't target it exclusively.

We also don't yet have a GWAPI-native approach to existing independent CRs (KongConsumers and global KongClusterPlugins), so long term we probably want to add a GWAPI-style attachment mechanism for all of those in one go.

Reply:
Done. Now it accepts both annotation and `spec.controllerName`.